pkg/util/ansible_execution.go

```diff
@@ -219,22 +220,25 @@ func AnsibleExecution(
 		util.LogErrorForObject(helper, err, fmt.Sprintf("Unable to create AnsibleEE %s", ansibleEE.Name), ansibleEE)
 		return err
 	}
-
+	util.LogForObject(helper, fmt.Sprintf("AnsibleEE %s created", ansibleEE.Name), ansibleEE)
 	return nil
 }
 
-// GetAnsibleExecution gets and returns an OpenStackAnsibleEE with the given
-// label where <label>=<node UID>
-// If none is found, return nil
-func GetAnsibleExecution(ctx context.Context, helper *helper.Helper, obj client.Object, label string) (*ansibleeev1.OpenStackAnsibleEE, error) {
+// GetAnsibleExecution gets and returns an OpenStackAnsibleEE with given
+// labels:
+// <label>=<deployment UID>
+// nodeset=<name of nodeset>
+// If multiple or no resources are found, return nil and error
+func GetAnsibleExecution(ctx context.Context, helper *helper.Helper, obj client.Object, label string, targetNodeSet string) (*ansibleeev1.OpenStackAnsibleEE, error) {
 	var err error
 	ansibleEEs := &ansibleeev1.OpenStackAnsibleEEList{}
 
 	listOpts := []client.ListOption{
 		client.InNamespace(obj.GetNamespace()),
 	}
 	labelSelector := map[string]string{
-		label: string(obj.GetUID()),
+		label:     string(obj.GetUID()),
+		"nodeset": targetNodeSet,
```

Comment:
can this be `openstackdataplanenodeset` since that is the actual name of the resource type, not `nodeset`

Reply:
do we want also want to go ahead and set the labels in a more consistent way? Instead of using a dynamic label based on service name, we could use:

```
"openstackdataplaneservice": <serviceName>,
"openstackdataplanenodeset": <nodeSetName>,
"openstackdataplanedeployment": <deploymentName>
```

Or use UID's instead of names?

Reply:
I think we should either use existing mix of name key and UID value. Or fully switch to names. Let's face it, names are more readable.

Reply:
I went with your suggestion.